app/models/game.rb

```diff
@@ -23,6 +24,91 @@ class Game < ActiveRecord::Base
 
   before_save :sync_current_inning_with_game_status, :prevent_mirror_matches
 
+  def next_half_inning
+    completed_half_innings = innings.completed.count
+
+    if innings.in_progress.count > 0
+      innings.in_progress.first
+    else
+      Inning.new(
+        game: self,
+        half: innings.completed.count % 2 == 0 ? :top : :bottom,
+        status: :in_progress,
+        number: next_inning(completed_half_innings),
+        runs: 0  
+      )
+    end
+  end
+
+  def going_final?
+    # TODO: Make this private and have this checked via callback
+    # call it check_finality or something
+    # collapse conditions if possible
+    # if someone wins it in the bottom of the 9th (etc), end it on that at bat!
+    return if self.final?
+    icc = innings.completed.count
+    if away_score == home_score
+      false
+    elsif icc < 17
+      false
+    elsif icc >= 17 && home_score > away_score
+      end_game
+      true
+    elsif icc == 18 && away_score > home_score && innings.count != 19
+      end_game 
+      true
+    elsif icc == 18 && away_score > home_score && innings.count == 19
+      false # keep playing
+    elsif icc % 2 != 0 && away_score > home_score
+      false # keep playing
+    elsif icc > 18 && home_score > away_score
+      end_game
+      true
+    elsif icc > 18 && icc % 2 == 0 && away_score > home_score
+      end_game
+      true
+    else
+      raise "Unreachable condition was reached"
+    end
+  end
+
+  def print_line_score
+    # TODO: Add Abbreviation to Team!!! (e.g. BOS, NYY)
```

Comment:
Added to `Team` but I still need to factor the new column into this method (i.e. remove hard coding of abbreviation here)